client/app/bundles/course/group/forms/GroupCreationForm.jsx

```diff
@@ -0,0 +1,245 @@
+  isSingle,
+  numToCreate,
+  name,
+  existingGroups,
+  error,
+}) => {
+  useEffect(() => {
+    dispatch(change(formNames.GROUP, 'is_single', true));
+  }, []);
+
+  const conflictingNames = useMemo(
+    () => getConflictingNames(name, numToCreate, existingGroups),
+    [name, numToCreate, existingGroups],
+  );
+
+  useEffect(() => {
+    if (!isSingle && numToCreate === conflictingNames.length) {
+      dispatch({ type: actionTypes.SET_IS_DISABLED_TRUE });
+    } else {
+      dispatch({ type: actionTypes.SET_IS_DISABLED_FALSE });
+    }
+  }, [numToCreate, conflictingNames]);
+
+  const handleChange = useCallback(
+    (value) =>
+      dispatch(change(formNames.GROUP, 'is_single', value === 'is_single')),
+    [dispatch],
+  );
+
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      <ErrorText errors={error} />
+      <Tabs
+        value={isSingle ? 'is_single' : 'is_multiple'}
+        onChange={handleChange}
+      >
+        <Tab label="Single" value="is_single">
+          <div style={styles.flexCol}>
+            <Field
+              name="name"
+              component={TextField}
+              floatingLabelText={<FormattedMessage {...translations.name} />}
+              disabled={submitting}
+              style={styles.flexChild}
+            />
+            <Field
+              name="description"
+              component={TextField}
+              floatingLabelText={
+                <FormattedMessage {...translations.description} />
+              }
+              multiLine
+              disabled={submitting}
+              rows={2}
+              rowsMax={4}
+              style={styles.flexChild}
+            />
+          </div>
+        </Tab>
+        <Tab label="Multiple" value="is_multiple">
+          <div style={styles.flexCol}>
+            <Field
+              name="name"
+              component={TextField}
+              floatingLabelText={<FormattedMessage {...translations.prefix} />}
+              disabled={submitting}
+              style={styles.flexChild}
+            />
+            <Field
+              name="num_to_create"
+              component={TextField}
+              floatingLabelText={
+                <FormattedMessage {...translations.numToCreate} />
+              }
+              type="number"
+              onWheel={(event) => event.currentTarget.blur()}
+              disabled={submitting}
+              style={styles.flexChild}
+              min={MIN_NUM_TO_CREATE}
+              max={MAX_NUM_TO_CREATE}
+            />
+            {name &&
+            numToCreate >= MIN_NUM_TO_CREATE &&
+            numToCreate <= MAX_NUM_TO_CREATE ? (
+              <div style={styles.note}>
+                This will create groups {name} 1 to {name} {numToCreate}.
+              </div>
+            ) : null}
+            {conflictingNames.length > 0 ? (
+              <div style={styles.warning}>
+                The following group(s) already exist and will not be created
```

Comment:
Dont forget to use Intl here.. I think not urgent as of now but put it in the TODO list.

Reply:
Thanks for the catch, fixed!